doc/nrf/libraries/others/trusted_storage.rst

```diff
@@ -0,0 +1,153 @@
+If this Kconfig option is set, the configuration defaults to the only currently available option :kconfig:option:`CONFIG_SECURE_STORAGE_STORAGE_BACKEND_SETTINGS` to use Zephyr's settings subsystem.
+
+The following options are used to configure the AEAD backend and its behavior:
+
+:kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND_AEAD_MAX_DATA_SIZE`
+   Defines the maximum data storage size for the AEAD backend (256 as default value).
+
+:kconfig:option:`CONFIG_SECURE_STORAGE_AEAD_BACKEND_CRYPTO`
+   Selects what implementation is used to perform the AEAD cryptographic operations.
+   This option defaults to :kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND_AEAD_CRYPTO_PSA_CHACHAPOLY` using the ChaCha20Poly1305 AEAD scheme via PSA APIs.
+
+:kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND_AEAD_NONCE`
+   Selects what implementation provides AEAD nonce.
+   You can choose one of the following values when this configuration option is set:
+
+   * :kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND_AEAD_NONCE_PSA_SEED_COUNTER` - Selects the PSA Crypto for nonce initial random seed and a counter incrementing nonce for each AEAD encryption.
+     This is the default option for the AEAD nonce.
+   * :kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND_AEAD_NONCE_CUSTOM` - Selects a custom implementation for AEAD nonce provider.
+
+:kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND_AEAD_KEY`
+   Selects what implementation provides the AEAD keys.
+   You can choose one of the following values when this configuration option is set:
+
+   * :kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK` - Selects HUK to derive a key based on the UID file.
+     This is the default selection for the AEAD key provider, and also the only really secure option.
+   * :kconfig:option:`CONFIG_SECURE_STORAGE_BACKEND_AEAD_KEY_HASH_UID` - Selects the use of SHA-256 of the UID file as the key.
+     This option is not as secure as when using HUKs for key derivation.
+     Use this option only when HUK is not possible to use.
+   * :kconfig:option:`CONFIG_SECURE_STORAGE_AEAD_BACKEND_KEY_CUSTOM` - Selects a custom implementation for the AEAD key provider.
+
+Usage
+*****
+
+The trusted storage library can only be used on a build using a build target with :ref:`CMSE disabled <app_boards_spe_nspe_cpuapp>` (``_cpuapp``).
+When you build for cpuapp, you build the firmware for the application core without CMSE and thus no TF-M.
+The library can be used directly on such a build to store important assets.
+However, for cryptographic keys we suggest to use the `PSA functions for key management`_.
+These APIs will internally use this library to store persistent keys.
+
+Dependencies
+************
+
+This library has dependencies to following libraries:
+
+* :ref:`lib_hw_unique_key`
+* :ref:`Zephyr's settings subsystem <zephyr:settings_api>`
+
+API documentation
+*****************
+
+Protected storage
+=================
+
+| Header file: :file:`include/protected_storage.h`
```

Comment:
We should use the headers in `subsys/encrypted_storage/include/psa/protected_storage.h` 
Instead of coping to the nrf global include folder

Reply:
Usually, to build the API docs, the header file is added to the include folder. Can the files be distinguished in any way, can they be named with ts_ prefix or something like that? Or is there a reason for not doing this.
There is a way of including the path to a specific file to be built, but I am checking if there are any rules/restrictions for its use.